src/main/java/ru/practicum/shareit/request/service/ItemRequestServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+        User requestor = getRequestorUser(userId);
+
+        Optional<ItemRequest> itemRequestOptional = requestRepository.findById(id);
+        if (itemRequestOptional.isPresent()) {
+            ItemRequest itemRequest = itemRequestOptional.get();
+
+            ItemRequestDto itemRequestDto = ItemRequestMapper.toItemRequestDto(itemRequest);
+            List<Item> itemList = itemRepository.findByRequestId(itemRequestDto.getId());
+            itemRequestDto.setItems(ItemMapper.toItemDtoList(itemList));
+
+            return itemRequestDto;
+        } else {
+            log.info("-------------Запрос не найден--------");
+            throw new IncorrectParameterException("Запрос не найден");
+        }
+    }
+
+    @Override
+    public List<ItemRequestDto> getItemRequests(Integer userId) {
+        User requestor = getRequestorUser(userId);
+        List<ItemRequest> itemRequestList = requestRepository.findByRequestor(requestor);
+        List<ItemRequestDto> itemRequestDtoList = ItemRequestMapper.toItemRequestDtoList(itemRequestList);
+        addItems(itemRequestDtoList);
+
+        itemRequestDtoList = sortItemRequestList(itemRequestDtoList);
+
+        return itemRequestDtoList;
+    }
+
+    @Override
+    public List<ItemRequestDto> getItemRequests(Integer owner, Integer from, Integer size) {
+        List<ItemRequest> itemRequestList = new ArrayList<>();
+        if (from == null || size == null) {
+            itemRequestList = requestRepository.findAll();
+            return ItemRequestMapper.toItemRequestDtoList(itemRequestList);
+        } else if (from < 0 || size <= 0) {
+            throw new IncorrectPageParametrException("Неверные параметры");
+        } else {
+//            Sort sortById = Sort.by(Sort.Direction.ASC, "id");
+//            Pageable page = PageRequest.of(from, size, sortById);
+            User requestor = getRequestorUser(owner);
+            itemRequestList = requestRepository.findByRequestorLimits(requestor.getId(), from, size);
+//            List<ItemRequestDto> itemRequestDtoList = null;
+//            if (itemRequestPage != null) {
+//                itemRequestList = itemRequestPage.getContent();
+            List<ItemRequestDto> itemRequestDtoList = ItemRequestMapper.toItemRequestDtoList(itemRequestList);
+            addItems(itemRequestDtoList);
+            itemRequestDtoList = sortItemRequestList(itemRequestDtoList);
+//            }
+            return itemRequestDtoList;
+        }
+
+    }
+
+    private List<ItemRequestDto> sortItemRequestList(List<ItemRequestDto> itemRequestDtoList) {
+        itemRequestDtoList = itemRequestDtoList.stream()
+                .sorted(Comparator.comparing(ItemRequestDto::getCreated))
+                .collect(Collectors.toList());
+        return itemRequestDtoList;
+    }
+
+    private List<ItemRequestDto> addItems(List<ItemRequestDto> itemRequestDtoList) {
+        itemRequestDtoList.stream()
```

Comment:
У списка уже есть встроенный метод forEach, можно удалить создание стрима:
```suggestion
        itemRequestDtoList
```